dmipy/utils/invariants.py

```diff
@@ -0,0 +1,576 @@
+    """
+    M = sh_matrix(sh_order, vecs, sh_base = sh_base)
+    L = _laplace_beltrami(sh_order) * lambda_lb
+    shape = data.shape
+    if len(shape) > 4:
+        raise ValueError(
+            'len(data.shape) must be lower than 5')
+    n_c = (sh_order + 1)*(sh_order + 2)//2
+    coef = np.zeros(list(shape[:-1])+[n_c])
+
+    if len(shape) == 1:
+        coef = _fit_sh(data, M, L)
+
+    if len(shape) == 2:
+        for i in range(shape[0]):
+            coef[i,:] = _fit_sh(data[i,:], M, L)
+
+    if len(shape) == 3:
+        for i in range(shape[0]):
+            for j in range(shape[1]):
+                coef[i,j,:] = _fit_sh(data[i,j,:], M, L)
+
+    if len(shape) == 4:
+        for i in range(shape[0]):
+            for j in range(shape[1]):
+                for k in range(shape[2]):
+                    coef[i,j,k,:] = _fit_sh(data[i,j,k,:], M, L)
+    return coef
+
+
+
+def gaunt_sph_harm(m,l,theta, phi):
+    r""" Compute spherical harmonics as defined in [1]
+    Parameters
+    ----------
+    m : int,
+        The order of the harmonic.
+    l : int,
+        The degree of the harmonic.
+    theta : float,
+        The azimuthal coordinate.
+    phi : float,
+        The polar coordinate.
+
+    Returns
+    --------
+    val : float,
+        The SH $y^m_l$ sampled at `theta` and `phi`.
+
+    See Also
+    --------
+    [1] Homeier, Herbert HH, and E. Otto Steinborn. 'Some properties of the
+    coupling coefficients of real spherical harmonics and their relation to
+    Gaunt coefficients.' Journal of Molecular Structure: THEOCHEM 368 (1996):
+    31-37.
+    """
+    x = np.cos(theta)
+    am = np.abs(m)
+    val = (1j)**(m+am)
+    if l+am == 0:
+        f = 1.0
+    else:
+        f= factorial(l-am)/factorial(l+am)
+
+    val*= np.sqrt((2*l+1)/(4*np.pi) * f).astype(complex)
+    val*= _LegendreP(am, l, x).astype(complex)
+    val*= np.exp(1j*m*phi)
+    return val
+
+
+def gaunt_real_sph_harm(m, l, theta, phi):
+    r""" Compute real spherical harmonics as defined in [1]
+    Parameters
+    ----------
+    m : int,
+        The order of the harmonic.
+    l : int,
+        The degree of the harmonic.
+    theta : float,
+        The azimuthal coordinate.
+    phi : float,
+        The polar coordinate.
+
+    Returns
+    --------
+    val : float,
+        The real SH $Y^m_l$ sampled at `theta` and `phi`.
+
+    See Also
+    --------
+    [1] Homeier, Herbert HH, and E. Otto Steinborn. 'Some properties of the
+    coupling coefficients of real spherical harmonics and their relation to
+    Gaunt coefficients.' Journal of Molecular Structure: THEOCHEM 368 (1996):
+    31-37.
+    """
+    if m>0:
+        val = np.sqrt(2) * np.real(gaunt_sph_harm(m,l,theta, phi))
+    if m==0:
+        val = np.real(gaunt_sph_harm(0,l,theta, phi))
+    if m<0:
+        val = np.sqrt(2) * np.imag(gaunt_sph_harm(np.abs(m),l,theta, phi))
+
+    return val
+
+def convert_sh(coef, in_base='mrtrix'):
```

Comment:
This function will just return a copy of `coef` when you misspell the `in_basis`. Add a catch.